CONTRIBUTING.md

```diff
@@ -74,29 +77,122 @@ make unit_testing
 ```
 
-### Smoke tests of cluster mode
 
-These tests check that cluster mode works as expected (`ray`, `vector-search`, `cdc`, `rest`):
+## Lint and type-check the code
 
-```shell
-make smoke_testing
+We use `black` for code formatting, `run` for linting, and `mypy` for type-checking.
+
+You can run the following commands to check the code:
+
+```
+make lint-and-type-check
+```
+
+If you want to format the code, you can run the following command:
 ```
+make fix-and-check
+```
+
+## Contributing to new plugins and templates
+
+The `superduper` project is designed to be extensible and customizable. You can create new plugins and templates to extend the functionality of the project.
+
+### Create a new plugin
+
+Plugins are Python packages that extend the functionality of the SuperDuper project.
+More details about the plugins can be found in the documentation:
+
+- [Data plugins](https://docs.superduper.io/docs/data_plugins/intro)
+- [AI plugins](https://docs.superduper.io/docs/category/ai-plugins)
+
+If you want to create a new plugin, you can follow the steps below:
+
+1. Copy `plugins/template` to `plugins/<PLUGIN_NAME>`
+2. Modify the name of the plugin in the following names:
+   - Plugin name in `pyproject.toml`
+   - Package name `superduper_<PLUGIN_NAME>`
+3. Write the code for the plugin
+4. Create the tests directory `plugin_test/` and write the tests
+
+### Create a new template
+
+Templates are reusable components that facilitate the quick and easy building of AI applications.
+More details about the templates can be found in the documentation:
+
+- [Templates](https://docs.superduper.io/docs/category/templates)
+
+If you want to create a new template, you can follow the steps below:
+
+1. Create a new directory in `templates/<TEMPLATE_NAME>`
+2. Create a `build.ipynb` file with the code to build the template
+   1. Present the build process in a Jupyter notebook
+   2. Package all the components into the application
+   3. Build a template from the application
+   4. Export the template
+
+## Contributing to the documentation
+We maintain the documentation in the [superduper-docs](https://github.com/superduper-io/superduper-docs) repository.
+
+
+Please go to the repository and create a pull request with the changes you want to make.
+
+### Fork and clone the repository
+
+```
+git clone git@github.com:<FORKED_NAME>/superduper-docs.git
+cd superduper-docs
+```
+
+### Updated the documentation
+
+For most document updates and additions, you can directly modify the files under `superduper-docs/content`.
+
+But there are some special cases:
+
+- Plugin documentation
+- Template documentation
+
+**Create or update a plugin documentation**
+
+After we create or update a plugin, we need to update the documentation.
+
+1. Update the `README.md` file in the plugin directory.
+2. Copy the file to the `superduper-docs/content/plugins` directory and change the file name to `<PLUGIN_NAME>.md`.
+
+**Create or update a template documentation**
```

Comment:
Creating of updating documentation for a template

Reply:
Done